pkg/reloader/reloader.go

```diff
@@ -692,21 +704,29 @@ func RuntimeInfoURLFromBase(u *url.URL) *url.URL {
 
 var envRe = regexp.MustCompile(`\$\(([a-zA-Z_0-9]+)\)`)
 
-func expandEnv(b []byte) (r []byte, err error) {
-	r = envRe.ReplaceAllFunc(b, func(n []byte) []byte {
+func (r *Reloader) expandEnv(b []byte) (replaced []byte, err error) {
+	replaced = envRe.ReplaceAllFunc(b, func(n []byte) []byte {
 		if err != nil {
 			return nil
 		}
+		m := n
 		n = n[2 : len(n)-1]
 
 		v, ok := os.LookupEnv(string(n))
 		if !ok {
-			err = errors.Errorf("found reference to unset environment variable %q", n)
+			r.configEnvVarExpansionErrors.Inc()
```

Comment:
you'd need to save the number of errors in a local variable and set the gauge before exiting expandEnv()

Reply:
Ah! So as to **not** take into account increments from a previous config expansion, right? I'll make the change.